vyper/ast/nodes.py

```diff
@@ -1086,6 +1106,36 @@ def evaluate(self) -> VyperNode:
         if not isinstance(self.op, (Eq, NotEq)) and not isinstance(left, (Int, Decimal)):
             raise TypeMismatch(f"Invalid literal types for {self.op.description} comparison", self)
 
+        # Simple typechecking
+        left_type = self.left._metadata.get("type", None)
+        right_type = self.right._metadata.get("type", None)
+        types = {left_type, right_type}
+
+        if None not in types and not left_type.compare_type(right_type):
+            raise UnfoldableNode(
+                f"Cannot perform {self.op._description} comparison between dislike types"
+            )
+
+        if None in types and len(types) == 2:
+            if left_type and not SizeLimits.in_bounds(repr(left_type), right.value):
+                # Check right value is in bounds
+                raise OverflowException(
+                    (
+                        f"The right literal's ({right.value}) is outside bounds of the ",
```

Comment:
i think we can assume in general that users are not super familiar with compiler jargon ("literal", etc.). what if we printed something like `"{right.value} cannot be {description} with {left.value} (which is of type {left_type})"`?